accepted/0000-hub_retail.md

```diff
@@ -0,0 +1,211 @@
+[design]: #detailed-design
+
+## Images + Image metadata
+
+Image files with corresponding metadata and pillars are already synchronized.
+Further customization is possible via API
+
+## HWTYPE configurations
+
+This is currently implemented as a formula assigned to a group with special name.
+
+FIXME: pick one alternative
+
+Alternative 1:
+Extend retail_yaml to create the groups and formula data via Hub and Peripheral Server API.
+This is relatively small change to the code.
+
+Alternative 2:
+Central creation of groups is covered in [Hub Users RFC](https://github.com/uyuni-project/uyuni-rfc/blob/master/accepted/00061-hub-users.md). The salt states
+can be extended to handle also formula data.
+This approach is probably more flexible.
+
+Both alternatives will have to cope with authentization and Hub vs. Peripheral server user credentials.
+
+In long term, alternative 2 is clean solution.
+
+Alternative 1 might be used as a short term fix and also to keep compatibility.
+
+
+### YAML configuration
+
+FIXME: optional, related to Alternative 2 in previous section
+
+We have retail_yaml tool for initial configuration of Retail environment from
+[YAML](https://github.com/uyuni-project/retail/blob/master/python-susemanager-retail/example.yml) file.
+[Hub Users RFC](https://github.com/uyuni-project/uyuni-rfc/blob/master/accepted/00061-hub-users.md) introduced similar concept, based on Salt.
+Maybe these parts could be merged.
+
+
+## Combined Peripheral and Branch server
+
+The components from Containerized Branch Server can be re-used also in this scenario.
+
+Required components:
+
+- Saltboot group
+- Cobbler / TFTP   
+- Mapping of branch-local URLs
+- DHCP server
+
+
+### Mapping of branch-local URLs
+
+https://github.com/SUSE/spacewalk/issues/18455
+
+Non-containerized retail branch server uses image pillars to map branch-local image URLs to SUMA server URLs.
+
+Containerized branch server does not have access to pillars, so it currently implements this 
+functionality by apache mod_rewrite. This is quite error-prone, because it expect URL in certain 
+format. It can't work with images on external server.
+https://github.com/SUSE/spacewalk/blob/Manager-4.3/containers/proxy-httpd-image/uyuni-configure.py#L166
+
+Proposed fix: implement the mapping by HTTP redirects in java on SUMA server. Java has access 
+to pillars in DB, so it does not have to do any guessing.
+This will cover also the Combined Peripheral and Branch server case.
+
+
+### DHCP server
+
+In Hub setup, DHCP on Peripheral Servers can be configured via formula on Hub.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we **not** do this?
+
+TBD
+
+# Alternatives
+[alternatives]: #alternatives
+
+HWTYPE configurations:
+
+TBD
+
+Combined Peripheral and Branch server:
+
+TBD
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- is nested Hub needed?
+  The use case with one Peripheral Server per shop might exceed the maximum number of Peripheral Servers.
```

Comment:
I'm not expecting that one peripheral per shop is main use case.
Even in that scenario, one hub could manage up to 10k suse manager servers, so It should be enough. If not, that is a problem I think everyone would be glad to solve :)